src/test/java/co/cask/cdc/plugins/sink/CDCBigTableConfigTest.java

```diff
@@ -0,0 +1,134 @@
+package co.cask.cdc.plugins.sink;
+
+import co.cask.cdap.etl.api.validation.InvalidConfigPropertyException;
+import com.google.bigtable.repackaged.com.google.cloud.ServiceOptions;
+import org.assertj.core.api.Assertions;
+import org.junit.Assume;
+import org.junit.Test;
+
+public class CDCBigTableConfigTest {
+  private static final String VALID_REF = "test-ref";
+  private static final String VALID_PROJECT = "test-project";
+  private static final String VALID_INSTANCE = "test-instance";
+  private static final String VALID_ACCOUNT_FILE_PATH
+    = CDCBigTableConfigTest.class.getResource("/credentials.json").getPath();
+
+  @Test
+  public void testValidateValidConfig() {
+    CDCBigTableConfig config = new CDCBigTableConfig(
+      VALID_REF,
+      VALID_INSTANCE,
+      VALID_PROJECT,
+      VALID_ACCOUNT_FILE_PATH
+    );
+
+    Assertions.assertThatCode(config::validate)
+      .doesNotThrowAnyException();
+  }
+
+  @Test
+  public void testValidateReference() {
+    CDCBigTableConfig config = new CDCBigTableConfig(
+      "",
+      VALID_INSTANCE,
+      VALID_PROJECT,
+      VALID_ACCOUNT_FILE_PATH
+    );
+
+    Assertions.assertThatThrownBy(config::validate)
```

Comment:
this should be replaced with a call to config.validate(), where you catch the exception and check that e.getProperty() is correct.

Reply:
same for the next few tests

Reply:
> just noticed that a lot of files are missing copyright. This is supposed to be in the rat check, have we not set it up for this project?

This plugin was not configured for this project as well as for google-cloud. I may submit another PR with rat plugin and updated copyrights in all classes.

Reply:
Added invalid property check. Can we use AssertJ here to have much more readable assertion and failure message? You can compare it with a code to make similar thing https://github.com/joel-costigliola/assertj-examples/blob/master/assertions-examples/src/test/java/org/assertj/examples/FailUsageExamples.java.
AssertJ also allows to easily extract test conditions to test framework to make tests for all plugins look similar and more readable. Currently there are no guides and multiple approaches.

Reply:
In general I don't think it's great to assert exception messages in tests as it is very brittle and tends to just test how things happen to be implemented rather than testing an explicit contract. 

That is why I suggested catching the InvalidConfigPropertyException and asserting that the property name is correct. That is an actual contract that the plugin must follow.

For other scenarios it may make sense to do what you suggested.

Reply:
I was trying to say about failure messages in test logs/reports, not about asserting exception messages